cmd/thanos/receive.go

```diff
@@ -369,7 +364,13 @@ func runReceive(
 		// watcher, we close the chan ourselves.
 		updates := make(chan receive.Hashring, 1)
 
-		if cw != nil {
+		// The Hashrings config file path given initializing config watcher.
+		if configPath, err := hashringsFile.Path(); err != nil {
```

Comment:
this logic seems backwards to me. If we receive an error getting the path then we continue and try to use the path? When `Path()` returns an error, the returned path is `""` so this will not work. Am I understanding it incorrectly?

Reply:
Thanks for catching this. Fixed the issue.